src/discordtipbot/discordbot/premium_cmd.cr

```diff
@@ -0,0 +1,43 @@
+class PremiumCmd
+  def initialize(@tip : TipBot)
+  end
+
+  def call(msg, ctx)
+    client = ctx[Discord::Client]
+    cmd = ctx[Command].command
+    cache = client.cache.not_nil!
+
+    begin
+      method = cmd[1]? if guild_id = cmd[0]?.try &.to_u64
+    rescue
+    end
+    guild_id = cache.resolve_channel(msg.channel_id).guild_id.try &.to_u64 unless guild_id
+    return unless guild_id
+
+    @tip.clear_expired_premium
+
+    case method || cmd[0]?
+    when "status"
+    when "extend"
+      time_array = method ? cmd[2..3] : cmd[1..2]
+      @tip.extend_premium(Premium::Kind::Guild, guild_id, time(time_array))
+    end
+
+    human = ctx[ConfigMiddleware].get_premium_string(Premium::Kind::Guild, guild_id)
+    string = human ? "The guild has premium for another **#{human}**" : "The guild does **not** have premium"
+    client.create_message(msg.channel_id, string)
+    yield
+  end
+
+  private def time(time_array) : Time::Span
+    number, span = time_array
+    case span
+    when .starts_with?("minute") then number.to_i.minutes
+    when .starts_with?("hour")   then number.to_i.hours
+    when .starts_with?("day")    then number.to_i.days
+    when .starts_with?("month")  then (number.to_i * 30).days
+    when .starts_with?("year")   then (number.to_i * 365).days
+    else                              0.seconds
```

Comment:
Currently `@tip.extend_premium(Premium::Kind::Guild, guild_id, 0.seconds)` is allowed, because of user error, or they actually pass `"0"`... does it make sense to allow this? I think it would be best to `raise` in else (or handle it, somehow) and give the user actual feedback that the time they passed in is invalid.

Maybe it would be easiest to validate both `number` and `span` with a regex up front in `call`, to address all of my above comments for this file. Up to you.

Reply:
I've opted to use a rescue. This will only ever be exposed to bot admins, so I don't think there's really any point in making this 100% foolproof